app/assets/javascripts/app/views/search_base_view.js

```diff
@@ -0,0 +1,124 @@
+app.views.SearchBase = app.views.Base.extend({
+  initialize: function(options){
+    this.ignoreDiasporaIds = [];
+    this.typeaheadInput = options.typeaheadInput;
+    this.setupBloodhound(options);
+    if(options.customSearch) { this.setupCustomSearch(); }
+    this.setupTypeahead();
+    // TODO: Remove this as soon as corejavascript/typeahead.js has its first release
+    this.setupMouseSelectionEvents();
+    if(options.autoselect) { this.setupAutoselect(); }
+  },
+
+  setupBloodhound: function(options) {
+    var bloodhoundOptions = {
+      datumTokenizer: function(datum) {
+        var nameTokens = Bloodhound.tokenizers.nonword(datum.name);
+        var handleTokens = datum.handle ? Bloodhound.tokenizers.nonword(datum.name) : [];
+        return nameTokens.concat(handleTokens);
+      },
+      queryTokenizer: Bloodhound.tokenizers.whitespace,
+      prefetch: {
+        url: "/contacts.json",
+        transform: this.transformBloodhoundResponse,
+        cache: false
+      },
+      sufficient: 5
+    };
+
+    // Allow bloodhound to look for remote results if there is a route given in the options
+    if(options.remoteRoute){
+      bloodhoundOptions.remote = {
+        url: options.remoteRoute + ".json?q=%QUERY",
+        wildcard: "%QUERY",
+        transform: this.transformBloodhoundResponse
+      };
+    }
+
+    this.bloodhound = new Bloodhound(bloodhoundOptions);
+  },
+
+  setupCustomSearch: function() {
+    var self = this;
+    this.bloodhound.customSearch = function(query, sync, async){
+      var _sync = function(datums){
+        var results = _.reject(datums, function(datum) {
```

Comment:
General note: Sometimes, there is a space between the closing parenthesis and the opening curly brace. Please unify. :)